esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,812 @@
+
+        if diag_config['write_plots']:
+            base_name_mod = (
+                'global_{title}_{source_label}_'
+                '{grid_label}'.format(
+                    title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                    source_label=key,
+                    grid_label=diag_config['obs_info']['grid_label']))
+            plot_path_mod = get_plot_filename(base_name_mod, diag_config)
+            plot_path_list.append(plot_path_mod)
+            _plot_single_map(plot_path_mod, tau_ctotal,
+                             global_tau_mod['global'][key], key, diag_config)
+
+    provenance_record = get_provenance_record(
+        "Comparison of global distributions of turnover time of carbon"
+        "with matrix comparison, maps, and multimodel bias. Reproduces"
+        "figure 3 in Carvalhais et al. (2014).", ['mean', 'perc'], ['global'],
+        _get_ancestor_files(diag_config, 'tau_ctotal'))
+
+    if diag_config['write_netcdf']:
+        model_cubes = [
+            c for c in global_tau_mod['grid'].values()
+            if isinstance(c, iris.cube.Cube)
+        ]
+        obs_cubes = [
+            c for c in global_tau_obs['grid'].values()
+            if isinstance(c, iris.cube.Cube)
+        ]
+        netcdf_path = get_diagnostic_filename(base_name, diag_config)
+        save_cubes = iris.cube.CubeList(model_cubes + obs_cubes)
+        iris.save(save_cubes, netcdf_path)
+    else:
+        netcdf_path = None
+
+    if diag_config['write_plots']:
+        # multimodel agreement
+        base_name_multimodel = '{prefix}_{base_name}'.format(
+            prefix='global_multimodelAgreement', base_name=base_name)
+        plot_path_multimodel = get_plot_filename(base_name_multimodel,
+                                                 diag_config)
+        _plot_multimodel_agreement(plot_path_multimodel, global_tau_mod,
+                                   global_tau_obs, config)
+        plot_path_list.append(plot_path_multimodel)
+
+        # map of observation
+        base_name_obs = '{prefix}_{base_name}'.format(prefix='global',
+                                                      base_name=base_name)
+        plot_path_obs = get_plot_filename(base_name_obs, diag_config)
+        _plot_single_map(plot_path_obs, global_tau_obs['grid']['tau_ctotal'],
+                         global_tau_obs['global']['tau_ctotal'],
+                         config['obs_info']['source_label'], config)
+        plot_path_list.append(plot_path_obs)
+
+        # matrix of maps
+        base_name_matrix = '{prefix}_{base_name}'.format(
+            prefix='global_matrix_map', base_name=base_name)
+        plot_path_matrix = get_plot_filename(base_name_matrix, diag_config)
+        _plot_matrix_map(plot_path_matrix, global_tau_mod, global_tau_obs,
+                         config)
+        plot_path_list.append(plot_path_matrix)
+
+        provenance_record['plot_file'] = ', '.join(plot_path_list)
```

Comment:
This will not work, a better solution would be to log a provenance record for each plot individually, e.g. something like:
```python
with ProvenanceLogger(diag_config) as provenance_logger:
    for plot_path in plot_path_list:
        provenance_logger.log(plot_path, provenance_record)
```
assuming the same preprocessor output files are used as input for each plot, is that the case?

Reply:
OK. I have added it in the loop as suggested.

Reply:
I had to revise it. Now, it writes the individual plot files in the log but does not open the logger many times. That was writing log files for each figure in the plots directory.